src/transport/smtp/connection_url.rs

```diff
@@ -112,7 +112,15 @@ pub(crate) fn from_connection_url<B: TransportBuilder>(connection_url: &str) ->
     }
 
     if let Some(password) = connection_url.password() {
-        let credentials = Credentials::new(connection_url.username().into(), password.into());
+        let percent_decode = |s: &str| {
+            percent_encoding::percent_decode_str(s)
+                .decode_utf8_lossy()
```

Comment:
Is lossy decoding the right choice here?

Reply:
You're right, it probably isn't as this is a configuration thing, so returning the error makes more sense.